src/main/java/view_post/use_case/ViewPostInteractor.java

```diff
@@ -0,0 +1,34 @@
+package view_post.use_case;
+
+public class ViewPostInteractor implements ViewPostInputBoundary {
+    private final ViewPostDsGateway dsGateway;
+    private final ViewPostOutputBoundary presenter;
+
+    /**
+     * Initialize the interactor for this use case.
+     *
+     * @param dsGateway The gateway to access the data repository.
+     * @param presenter The presenter of this use case.
+     */
+    public ViewPostInteractor(ViewPostDsGateway dsGateway, ViewPostOutputBoundary presenter) {
+        this.dsGateway = dsGateway;
+        this.presenter = presenter;
+    }
+
+    /**
+     * Display the selected post.
+     * @param requestModel The input data from the user.
+     */
+    @Override
+    public void displayPost(ViewPostRequestModel requestModel) {
+        int postID = requestModel.getPostID();
+        String[] postInfo = dsGateway.getPostInfo(postID);
+        String delimiter = "##";
```

Comment:
I could be wrong, but I feel that the delimiter stuff should be in the view post gateway. Mainly just because the delimiter stuff is concerned with the specifics of the database, and we want to hide that from the interactor.

Reply:
I was thinking that the gateway's job is to just parse the database and get the post info and return it as an array of strings. I was planning on making the DS gateway parse it and add the delimiter so I can turn it into an array of strings in this method.
It's to avoid having the `getPostInfo` method return an array of objects and typecasting instead of just an array of strings.